cmd/kubeadm/app/images/images.go

```diff
@@ -46,6 +47,9 @@ func GetDNSImage(cfg *kubeadmapi.ClusterConfiguration) string {
 	if cfg.DNS.ImageRepository != "" {
 		dnsImageRepository = cfg.DNS.ImageRepository
 	}
+	if dnsImageRepository == kubeadmapiv1beta2.DefaultImageRepository {
```

Comment:
Could we add a small comment for future us to refer on why this is needed?

Reply:
Do we need to check that the version here is greater than 1.8.0? @fabriziopandini @neolit123

Reply:
@ykakarap mentioned that old images now also exist under k8s.gcr.io/coredns/coredns:oldtag.

Reply:
Ah ok, that works!

Reply:
I was digging into this and it seems that kubeadm v1.22 always ships coreDNS v1.8.0
The same is for kubeadm v1.21 when we backport this
@neolit123 to confirm that

Reply:
> @ykakarap mentioned that old images now also exist under k8s.gcr.io/coredns/coredns:oldtag

Just observed a thing to note here: the new path of the 1.7.0 image is `k8s.gcr.io/coredns/coredns:v1.7.0` and the path without the subpath is `k8s.gcr.io/coredns:1.7.0`. Notice the difference in the tag (one of them has `v`). We should probably handle that case.

Reply:
we don't have to check for version, IMO.
we can have a comment on top of the check as @vincepri suggested:

> Could we add a small comment for future us to refer on why this is needed?

Reply:
@ykakarap the "v" is embedded in https://github.com/kubernetes/kubernetes/blob/e5b54d07695a6066b00d6c837ebd584c76da8a00/cmd/kubeadm/app/constants/constants.go#L328
So we should be ok

+1 for the comment; starting from what we have in CAPI it should be something on the lines of "// Handle the renaming of the official image from "k8s.gcr.io/coredns" to "k8s.gcr.io/coredns/coredns"

Reply:
@ykakarap expanding a little bit, kubeadm v1.22 and kubeadm v1.21 will never ship CoreDNS v1.7.0, so we should not care about the image tag without the v